src/types.rs

```diff
@@ -0,0 +1,173 @@
+use super::sys;
+use std::{ffi::CStr, os::raw::c_char, time::Duration};
+
+fn bytes_to_cstr(bytes: &[c_char]) -> &CStr {
+    unsafe { CStr::from_ptr(bytes.as_ptr()) }
```

Comment:
Could we use `CStr::from_bytes_until_nul(...).expect("there is a terminating null")`? Just out of paranoia, so if they've done something dumb it will stop traversing at the known length and produce a good panic instead of Undefined Behavior.

Reply:
sounds like a job for 🦸 transmute 🦹 !